### ISSUE TITLE
RTopic’s subscription function always fails randomly

### ISSUE THREAD

**Issue by tzfun** [redisson/redisson#5709]
**Expected behavior**
Subscribe to 10 channels/topics, and then publish messages to these 10 channels. There are always 1-2 topics that cannot consume messages, and they are not fixed, but random channels.

**Steps to reproduce or test case**
Do nothing on the basis of the code, only upgrade the Redisson version. Problems will occur when upgrading from 3.25.1 to 3.27.2 or 3.26.1. When the version is changed back to 3.25.1, the problem will disappear.

**Redis version**
7.2.1

**Redisson version**
3.27.2 or 3.26.1

**Redisson configuration**
unimportant

**Comment by mrniko** [comment]
Unable to reproduce with the test below. What did you I miss?

```java
        int loops = 10;
        AtomicInteger counter = new AtomicInteger();
        for (int j = 0; j < loops; j++) {
            RTopic t = redisson.getTopic("PUBSUB_" + j);
            t.addListener(String.class, new MessageListener<String>() {
                @Override
                public void onMessage(CharSequence channel, String msg) {
                    System.out.println("channel " + channel + " " + msg);
                    counter.incrementAndGet();
                }
            });
        }

        for (int j = 0; j < loops; j++) {
            RTopic t = redisson.getTopic("PUBSUB_" + j);
            for (int i = 0; i < 10; i++) {
                t.publish("message " + j + "_" + i);
            }
        }

        Thread.sleep(1000);

        assertThat(counter.get()).isEqualTo(loops * 10);
```

**Comment by tzfun** [comment]
It is not easy to reproduce in a short period of time. You need to continue publishing messages, and then after running for a period of time, you will find that several channels cannot be consumed.

> Unable to reproduce with the test below. What did I miss?

**Comment by tzfun** [comment]
I don't know the specific reason. In my scenario: the same redis, the same number of channels, the same publish, the same configuration, and different Redisson versions.

I once suspected that the redis buffer overflow caused the client connection to be disconnected. I used the `client list` command to check the buffer usage and found that the buffer was very sufficient, thus ruling out the possibility of the redis server actively disconnecting.

Then I tried to add the Status listener of the Topic, and found that when the count bit of the publish result was 0, the status listener did not trigger any unsubscribe event. I tried repeatedly by switching different Redisson versions, and found that only 3.25.1 had no problem, so I had to Roll back to previous version.

**Comment by mrniko** [comment]
Can you try 3.26.0?

**Comment by tzfun** [comment]
> Can you try 3.26.0?

I can test it later when I have time, but I'm sorry not now.

**Comment by mrniko** [comment]
do invoke RTopic.addListener() only once per channel? Total 10 channels?

**Comment by tzfun** [comment]
This is my code. Each channel only triggers this code once.

![IMG_0185](https://github.com/redisson/redisson/assets/34815354/466ddaaf-0f38-492b-ab5f-57039202f2f5)

**Comment by tzfun** [comment]
This is my code. Each channel only triggers this code once.

```java
        RTopic rt = getTopic(topic);

        rt.addListener(new StatusListener() {
            @Override
            public void onSubscribe(String channel) {
                logger.debug("redis channel subscribed {}", channel);
            }

            @Override
            public void onUnsubscribe(String channel) {
                logger.info("redis channel unsubscribed {}", channel);
            }
        });

        rt.addListener(
                TransformMessage.class,
                (channel, msg) -> executor.execute(() -> consume(MQMessage.create(channel.toString(), msg), consumer))
        );
```

**Comment by mrniko** [comment]
Do you use a single Redis server or cluster setup?

**Comment by mrniko** [comment]
Fixed in https://github.com/redisson/redisson/pull/6244
